### PR TITLE
unit testing for handler, service and dao modules

### PR DESCRIPTION
**Pull request by ashwinspg**

### CODE REVIEW

File: daos/movies_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/ashwinspg/explore-golang/dtos"
+	"github.com/ashwinspg/explore-golang/test"
+)
+
```
Comment (suryapandian) [inline daos/movies_test.go:11]: imports we usually follow this order:
1. inbuilt packages
2.  current repo packages
3.  third party libraries packages

So, the import will be

```suggestion
	"testing"
	
	"github.com/ashwinspg/explore-golang/dtos"
	"github.com/ashwinspg/explore-golang/test"

	"github.com/stretchr/testify/suite"	
```

File: handlers/movies_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package handlers
+
+import (
```
Comment (suryapandian) [inline handlers/movies_test.go:3]: Please fix import order

File: handlers/movies_test.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+func TestGetMovie(t *testing.T) {
+	test.SetupTestEnv()
+	request := httptest.NewRequest(http.MethodGet, "/movies/0815d6a2-67ba-4487-a529-142f28f4d21c", nil)
+	w := httptest.NewRecorder()
+	GetMovieHandler(w, request)
+	response := w.Result()
+	fmt.Println(response)
+	assert.New(t).Equal(http.StatusOK, response.StatusCode)
+}
```
Comment (suryapandian) [inline handlers/movies_test.go:20]: Super 👏

File: services/movies_test.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package services
+
+import (
```
Comment (suryapandian) [inline services/movies_test.go:3]: Please fix import order.

File: services/movies_test.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+func (t *MovieTestSuite) TestGetMovie() {
+	movieDTO, err := t.service.GetMovie("1d3095da-3243-4a34-a7d6-2cb570446ffe")
+	t.NotNil(movieDTO)
+	t.NoError(err)
+
+	movieDTO, err = t.service.GetMovie("1d3095da-3243-4a34-a7d6-2cb570446fff")
+	t.Equal(dtos.Movie{}, movieDTO)
+	t.Error(err)
+}
```
Comment (suryapandian) [inline services/movies_test.go:33]: Can try adding test cases for the errors that have been handled in services as well.

File: test.env [hunk 6]
```diff
@@ -0,0 +1 @@
+export POSTGRES_DB_NAME=explore_golang_test
```
Comment (suryapandian) [inline test.env]: can just be DB_NAME 🤔

File: test/test_init.go [hunk 7]
```diff
@@ -0,0 +1,32 @@
+package test
+
+import (
```
Comment (suryapandian) [inline test/test_init.go:3]: Please fix import order

File: test/test_init.go [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+var once sync.Once
+
+//Env holds a test environment required to run the tests
+type Env struct {
+	DBConn *sql.DB
+	L      *logrus.Entry
+}
+
+//SetupTestEnv initialises and returns a complete test environment
+func SetupTestEnv() Env {
```
Comment (suryapandian) [inline test/test_init.go:22]: Nicely done 👏

**Review by suryapandian — CHANGES_REQUESTED** [review]

File: daos/movies_test.go (hunk 1 above)
Comment (ashwinspg) [reply daos/movies_test.go:11]: I tried to follow the order:

1. inbuilt packages
2. third party libraries packages
3. current repo packages

I'll change as you suggested.

File: test.env (hunk 6 above)
Comment (ashwinspg) [reply test.env]: sure, I'll make the change

File: services/moviebuff_mock.go [hunk 9]
```diff
@@ -0,0 +1,41 @@
+package services
```
Comment (suryapandian) [inline services/moviebuff_mock.go:1]: We could add mock if and when we are making use of mock in a separate PR or so.

**Review by suryapandian — APPROVED** [review]

**Review by suryapandian — APPROVED** [review]
Well done 👏

**Review by svetha-cvl — APPROVED** [review]
🚀 Nicely done!!

**Pull request merged**